src/web/lib/components/ThemeColorsEditor/index.js

```diff
@@ -54,16 +61,25 @@ class ThemeColorsEditor extends React.Component {
   render() {
     const {
       theme: { colors },
-      selectedColor
+      selectedColor,
+      advancedColors,
+      hasExtension,
+      extensionVersion,
+      addonUrl
     } = this.props;
 
+    const labels = advancedColors ? advancedColorLabels : colorLabels;
+
+    let requiredVersion = "2.1.6";
+    const updatedExtensionVersion = extensionVersion && semverCompare(extensionVersion, requiredVersion) >= 0;
+
     // Select only the color properties from the theme.
-    const colorKeys = Object.keys(colors).filter(name => name in colorLabels);
+    const colorKeys = Object.keys(labels);
 
     // Dedupe colors for swatch presets
     const uniqueColorArray = [
       ...new Set(
-        colorKeys.map(name => {
+        Object.keys(Object.assign({}, colorLabels, advancedColorLabels)).map(name => {
```

Comment:
This is not wrong, but since the use of spreads are widespread in the code base, how about using it here too?

`{...colorLabels, ...advancedColorLabels}`